src/ports/STM32Cube/sampleapp_main.c

```diff
@@ -0,0 +1,115 @@
+/*********************************************************************
+ *        _       _         _
+ *  _ __ | |_  _ | |  __ _ | |__   ___
+ * | '__|| __|(_)| | / _` || '_ \ / __|
+ * | |   | |_  _ | || (_| || |_) |\__ \
+ * |_|    \__|(_)|_| \__,_||_.__/ |___/
+ *
+ * www.rt-labs.com
+ * Copyright 2021 rt-labs AB, Sweden.
+ *
+ * This software is dual-licensed under GPLv3 and a commercial
+ * license. See the file LICENSE.md distributed with this software for
+ * full license information.
+ ********************************************************************/
+
+#include "sampleapp_common.h"
+#include "app_utils.h"
+#include "app_log.h"
+#include "app_gsdml.h"
+
+#include "osal_log.h"
+#include "osal.h"
+#include <pnet_api.h>
+
+#include <lwip/netif.h>
+
+#include <string.h>
+
+#define APP_DEFAULT_ETHERNET_INTERFACE "st"
+#define APP_DEFAULT_FILE_DIRECTORY     ""
+#define APP_LOG_LEVEL                  APP_LOG_LEVEL_INFO
+
+#define APP_BG_WORKER_THREAD_PRIORITY  2
+#define APP_BG_WORKER_THREAD_STACKSIZE 4096 /* bytes */
+
+/********************************** Globals ***********************************/
+
+static app_data_t * sample_app = NULL;
+static pnet_cfg_t pnet_cfg = {0};
+app_args_t app_args = {0};
+
+/****************************** Main ******************************************/
+
+int _main (void)
+{
+   int ret;
+   int32_t app_log_level = APP_LOG_LEVEL;
+   app_utils_netif_namelist_t netif_name_list;
+   pnet_if_cfg_t netif_cfg = {0};
+   uint16_t number_of_ports;
+
+   strcpy (app_args.eth_interfaces, APP_DEFAULT_ETHERNET_INTERFACE);
+   strcpy (app_args.station_name, APP_GSDML_DEFAULT_STATION_NAME);
+   app_log_set_log_level (app_log_level);
+
+   APP_LOG_INFO ("\n** Starting P-Net sample application " PNET_VERSION
+                 " **\n");
+   APP_LOG_INFO (
+      "Number of slots:      %u (incl slot for DAP module)\n",
+      PNET_MAX_SLOTS);
+   APP_LOG_INFO ("P-net log level:      %u (DEBUG=0, FATAL=4)\n", LOG_LEVEL);
+   APP_LOG_INFO ("App log level:        %u (DEBUG=0, FATAL=4)\n", app_log_level);
+   APP_LOG_INFO ("Max number of ports:  %u\n", PNET_MAX_PHYSICAL_PORTS);
+   APP_LOG_INFO ("Network interfaces:   %s\n", app_args.eth_interfaces);
+   APP_LOG_INFO ("Default station name: %s\n", app_args.station_name);
+
+   app_pnet_cfg_init_default (&pnet_cfg);
+
+   /* Note: station name is defined by app_gsdml.h */
+
+   strcpy (pnet_cfg.file_directory, APP_DEFAULT_FILE_DIRECTORY);
+
+   /* Note: pnal_cfg not is used for rt-kernel  */
```

Comment:
Update or remove comment?

Reply:
This was copied from rt-kernel/sampleapp_main.c. Will remove in both files.